core/src/main/java/org/javarosa/core/model/FormDef.java

```diff
@@ -854,6 +876,35 @@ private void initializeTriggerables(TreeReference rootRef) {
     }
 
     /**
+     * Evaluate triggerables targeting references that are children of the
+     * provided newly create repeat instance ref.  Ignore all triggerables that
+     * were already fired by processing the jr-insert action.
+     *
+     * @param newRepeatRef refers to newly inserted repeat entry
+     */
+    private void initializeNewRepeatItemTriggerables(TreeReference newRepeatRef) {
+        TreeReference genericRoot = newRepeatRef.genericize();
+
+        Vector<Triggerable> applicable = new Vector<Triggerable>();
+        for (int i = 0; i < triggerables.size(); i++) {
+            Triggerable t = (Triggerable)triggerables.elementAt(i);
+            for (int j = 0; j < t.getTargets().size(); j++) {
+                TreeReference target = (TreeReference)t.getTargets().elementAt(j);
+                if (genericRoot.isParentOf(target, false)) {
+                    if (!triggerablesProcessedInSetAction.contains(t)) {
+                        applicable.addElement(t);
+                    }
```

Comment:
Doesn't add triggerable for evaluation if it is was processed by the insert action stage.